rewrite-core/src/main/java/org/openrewrite/text/FindAndReplace.java

```diff
@@ -156,6 +192,15 @@ public Tree visit(@Nullable Tree tree, ExecutionContext ctx) {
 
             visitor = Preconditions.check(check, visitor);
         }
+
+        if (Boolean.TRUE.equals(plaintextOnly)) {
+            visitor = Preconditions.check(new PlainTextVisitor<ExecutionContext>(){
```

Comment:
I hadn't seen this pattern of effectively nested `Precondition.check` before, as opposed to and/or-ed together check condition; not sure how I feel about those yet. I'd have to look a little closer in my IDE.

Reply:
Looks ok; or at least unlikely to conflict given that one is a file pattern already, and folks are likely to use one or the other.